Compile bug: `error: ‘svr’ was not declared in this scope` after #17216

### Git commit

0de8878c961c36c88cca5a0c188ae7a6e5a4b778

### Operating systems

Linux

### GGML backends

CUDA, CPU

### Problem description & steps to reproduce

Since #17216 was merged, compilation of `llama-server` fails.

### First Bad Commit

0de8878c961c36c88cca5a0c188ae7a6e5a4b778

### Compile command

```shell
rm -rf build
time cmake -B build -DGGML_CUDA=1 -DCMAKE_CUDA_ARCHITECTURES="89" -DLLAMA_OPENSSL=1 -DGGML_CUDA_FA_ALL_QUANTS=1 # -DGGML_CCACHE=0 -DCMAKE_BUILD_TYPE=Debug
time cmake --build build --config Release -j 16
```

### Relevant log output

```shell
/home/dylan/llama.cpp/tools/server/server-http.cpp: In member function ‘bool server_http_context::init(const common_params&)’:
/home/dylan/llama.cpp/tools/server/server-http.cpp:52:9: error: ‘svr’ was not declared in this scope
   52 |         svr.reset(
      |         ^~~
/home/dylan/llama.cpp/tools/server/server-http.cpp:57:9: error: ‘svr’ was not declared in this scope
   57 |         svr.reset(new httplib::Server());
      |         ^~~
gmake[2]: *** [tools/server/CMakeFiles/llama-server.dir/build.make:98: tools/server/CMakeFiles/llama-server.dir/server-http.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
[100%] Linking CXX executable ../bin/test-backend-ops
[100%] Built target test-backend-ops
gmake[1]: *** [CMakeFiles/Makefile2:4019: tools/server/CMakeFiles/llama-server.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
```

--- a/tools/server/server-http.cpp
+++ b/tools/server/server-http.cpp
@@ -46,25 +46,26 @@ bool server_http_context::init(const common_params & params) {
     port = params.port;
     hostname = params.hostname;
 
+    auto & srv = pimpl->srv;
+
 #ifdef CPPHTTPLIB_OPENSSL_SUPPORT
     if (params.ssl_file_key != "" && params.ssl_file_cert != "") {
         LOG_INF("Running with SSL: key = %s, cert = %s\n", params.ssl_file_key.c_str(), params.ssl_file_cert.c_str());
-        svr.reset(
+        srv.reset(
             new httplib::SSLServer(params.ssl_file_cert.c_str(), params.ssl_file_key.c_str())
         );
     } else {
         LOG_INF("Running without SSL\n");
-        svr.reset(new httplib::Server());
+        srv.reset(new httplib::Server());
     }
 #else
     if (params.ssl_file_key != "" && params.ssl_file_cert != "") {
         LOG_ERR("Server is built without SSL support\n");
         return false;
     }
-    pimpl->srv.reset(new httplib::Server());
+    srv.reset(new httplib::Server());
 #endif
 
-    auto & srv = pimpl->srv;
     srv->set_default_headers({{"Server", "llama.cpp"}});
     srv->set_logger(log_server_request);
     srv->set_exception_handler([](const httplib::Request &, httplib::Response & res, const std::exception_ptr & ep) {

